framework/src/multiapps/MultiApp.C

```diff
@@ -357,13 +365,35 @@ MultiApp::createApps()
                                 getParam<std::string>("library_name"),
                                 getParam<bool>("library_load_dependencies"));
 
+  bool rank_did_quiet_init = false;
+  unsigned int local_app;
```

Comment:
```suggestion
  unsigned int local_app = libMesh::invalid_uint;
```
I don't know if valgrind would warn about this (I've added that recipe), but it makes me a little worried